FinSearchUnified/Tests/Bundle/SearchBundleFindologic/QueryBuilderFactoryTest.php

```diff
@@ -0,0 +1,282 @@
+        $this->assertSame('salesfrequency ASC', $params['order'], 'Expected sorting order to be "salesfrequency ASC"');
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function testCreateProductQueryWithoutConditionsAndSortings()
+    {
+        $criteria = new Criteria();
+
+        $query = $this->factory->createProductQuery($criteria, $this->context);
+        $params = $query->getParameters();
+
+        $this->assertArrayNotHasKey('order', $params, 'Sorting was not expected to be present in the parameters');
+        $this->assertArrayNotHasKey('attrib', $params, 'No attributes were expected to be present in the parameters');
+        $this->assertArrayNotHasKey('query', $params, 'No search query was expected to be present in the parameters ');
+    }
+
+    /**
+     * @dataProvider sortingProvider
+     *
+     * @param SortingInterface $sorting
+     * @param string $expected
+     *
+     * @throws Exception
+     */
+    public function testCreateProductQueryWithSingleSorting(SortingInterface $sorting, $expected)
+    {
+        $criteria = new Criteria();
+        $criteria->addSorting($sorting);
+
+        $query = $this->factory->createProductQuery($criteria, $this->context);
+        $params = $query->getParameters();
+
+        if ($expected === null) {
+            $this->assertArrayNotHasKey('order', $params, 'Did not expect order to exist in the parameters');
+        } else {
+            $this->assertArrayHasKey('order', $params, 'Did not expect order to exist in the parameters');
+            $this->assertSame($expected, $params['order'], sprintf('Expected sorting to be "%s"', $expected));
+        }
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function testCreateProductQueryWithMultipleSortingAndConditions()
+    {
+        $criteria = new Criteria();
+        $criteria->addSorting(new PopularitySorting());
+        $criteria->addSorting(new SimpleSorting('name'));
+        $criteria->addCondition(new SearchTermCondition('blubbergurke'));
+        $criteria->addCondition(new IsAvailableCondition());
+
+        $query = $this->factory->createProductQuery($criteria, $this->context);
+        $params = $query->getParameters();
+
+        $this->assertArrayHasKey('query', $params, 'Search query was expected to be present in the parameters');
+        $this->assertSame('blubbergurke', $params['query'], 'Expected "blubbergurke" to be the search query');
+        $this->assertArrayNotHasKey('attrib', $params, 'No attributes were expected to be present in the parameters');
+
+        $this->assertArrayHasKey('order', $params, 'Sorting was expected to be present in the parameters');
+        $this->assertSame('salesfrequency ASC', $params['order'], 'Expected sorting order to be "salesfrequency ASC"');
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function testCreateProductQueryWithOffsetAndLimit()
+    {
+        $criteria = new Criteria();
+        $criteria->offset(5);
```

Comment:
What about the case where offset is 0 and limit 1?

Reply:
you said to remove those test cases as we have setup the condition to make the limit 0 on this condition

Reply:
I said that you can drop the case where limit was supposed to be zero since the min value for it is 1. `offset = 0` and `limit = 1` is still a valid test case.

Reply:
I see, okay